test/integration/consul-container/test/upgrade/l7_traffic_management/traffic_management_test.go

```diff
@@ -0,0 +1,140 @@
+//
+// Steps:
+//   - Create a single agent cluster.
+//   - Create one static-server and 2 subsets and 1 client and sidecar, then register them with Consul
+//   - Validate static-server and 2 subsets are and proxy admin endpoint is healthy
+//   - Validate static servers proxy listeners should be up and have right certs
+func TestTrafficManagement_SetupServerAndClientWithSubsets(t *testing.T) {
+	t.Parallel()
+
+	type testcase struct {
+		oldversion    string
+		targetVersion string
+	}
+	tcs := []testcase{
+		{
+			oldversion:    "1.14",
+			targetVersion: utils.TargetVersion,
+		},
+	}
+
+	run := func(t *testing.T, tc testcase) {
+		cluster, _, _ := topology.NewPeeringCluster(t, "dc1", 1, tc.oldversion)
+
+		// Register service resolver
+		serviceResolver := &api.ServiceResolverConfigEntry{
+			Kind:          api.ServiceResolver,
+			Name:          libservice.StaticServerServiceName,
+			DefaultSubset: "v2",
+			Subsets: map[string]api.ServiceResolverSubset{
+				"v1": {
+					Filter: "Service.Meta.version == v1",
+				},
+				"v2": {
+					Filter: "Service.Meta.version == v2",
+				},
+			},
+		}
+		err := cluster.ConfigEntryWrite(serviceResolver)
+		require.NoError(t, err)
+
+		serverService, serverServiceV1, serverServiceV2, clientService := createService(t, cluster)
+
+		_, port := clientService.GetAddr()
+		_, adminPort := clientService.GetAdminAddr()
+		_, serverAdminPort := serverService.GetAdminAddr()
+		_, serverAdminPortV1 := serverServiceV1.GetAdminAddr()
+		_, serverAdminPortV2 := serverServiceV2.GetAdminAddr()
+
+		// validate client and proxy is up and running
+		libassert.AssertContainerState(t, clientService, "running")
+		libassert.HTTPServiceEchoes(t, "localhost", port, "")
+		libassert.AssertUpstreamEndpointStatus(t, adminPort, "v2.static-server.default", "HEALTHY", 1)
+
+		// Upgrade cluster and begin service validation
+		require.NoError(t, cluster.StandardUpgrade(t, context.Background(), tc.targetVersion))
+
+		// POST upgrade validation; repeat client & proxy validation
+		libassert.HTTPServiceEchoes(t, "localhost", port, "")
+		libassert.AssertUpstreamEndpointStatus(t, adminPort, "v2.static-server.default", "HEALTHY", 1)
+
+		// validate static-server, static-server-v1, static-server-v2 and static-client
+		// proxy admin are up
+		libassert.AssertServiceProxyAdminStatus(t, "localhost", adminPort)
+		libassert.AssertServiceProxyAdminStatus(t, "localhost", serverAdminPort)
+		libassert.AssertServiceProxyAdminStatus(t, "localhost", serverAdminPortV1)
+		libassert.AssertServiceProxyAdminStatus(t, "localhost", serverAdminPortV2)
+
+		// certs are valid
+		libassert.AssertEnvoyPresentsCertURI(t, adminPort, "static-client")
+		libassert.AssertEnvoyPresentsCertURI(t, serverAdminPort, "static-server")
+		libassert.AssertEnvoyPresentsCertURI(t, serverAdminPortV1, "static-server")
+		libassert.AssertEnvoyPresentsCertURI(t, serverAdminPortV2, "static-server")
+
+		// TO-DO: validate traffic management
```

Comment:
How how rephrase the TODO like "restart envoy sidecar and validate traffic management"